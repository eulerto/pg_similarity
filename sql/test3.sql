--
-- pg_similarity
-- testing similarity functions and operators
--

--
-- Clean up in case a prior regression run failed
--
RESET client_min_messages;
\set ECHO all

\set a '\'Euler Taveira de Oliveira\''

CREATE TABLE simtst (a text);

INSERT INTO simtst (a) VALUES
('Euler Taveira de Oliveira'),
('EULER TAVEIRA DE OLIVEIRA'),
('Euler T. de Oliveira'),
('Oliveira, Euler T.'),
('Euler Oliveira'),
('Euler Taveira'),
('EULER TAVEIRA OLIVEIRA'),
('Oliveira, Euler'),
('Oliveira, E. T.'),
('ETO');

-- Levenshtein
SELECT a FROM simtst WHERE a ~== :a;
SHOW pg_similarity.levenshtein_threshold;
SET pg_similarity.levenshtein_threshold to 0.4;
SHOW pg_similarity.levenshtein_threshold;
SELECT a FROM simtst WHERE a ~== :a;

-- Cosine
SHOW pg_similarity.cosine_threshold;
SELECT a FROM simtst WHERE a ~## :a;
SET pg_similarity.cosine_threshold to 0.9;
SHOW pg_similarity.cosine_threshold;
SELECT a FROM simtst WHERE a ~## :a;

-- Overlap Coefficient
SHOW pg_similarity.overlap_tokenizer;
SELECT a FROM simtst WHERE a ~** :a;
SET pg_similarity.overlap_tokenizer to 'gram';
SET pg_similarity.overlap_threshold to 0.8;
SELECT a FROM simtst WHERE a ~** :a;

DROP TABLE simtst;
