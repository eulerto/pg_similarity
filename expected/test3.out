--
-- pg_similarity
-- testing similarity functions and operators
--
--
-- Clean up in case a prior regression run failed
--
RESET client_min_messages;
\set ECHO all
\set a '\'Euler Taveira de Oliveira\''
CREATE TABLE simtst (a text);
INSERT INTO simtst (a) VALUES
('Euler Taveira de Oliveira'),
('EULER TAVEIRA DE OLIVEIRA'),
('Euler T. de Oliveira'),
('Oliveira, Euler T.'),
('Euler Oliveira'),
('Euler Taveira'),
('EULER TAVEIRA OLIVEIRA'),
('Oliveira, Euler'),
('Oliveira, E. T.'),
('ETO');
-- Levenshtein
SELECT a FROM simtst WHERE a ~== :a;
             a             
---------------------------
 Euler Taveira de Oliveira
 EULER TAVEIRA DE OLIVEIRA
 Euler T. de Oliveira
 EULER TAVEIRA OLIVEIRA
(4 rows)

SHOW pg_similarity.levenshtein_threshold;
 pg_similarity.levenshtein_threshold 
-------------------------------------
 0.7
(1 row)

SET pg_similarity.levenshtein_threshold to 0.4;
SHOW pg_similarity.levenshtein_threshold;
 pg_similarity.levenshtein_threshold 
-------------------------------------
 0.4
(1 row)

SELECT a FROM simtst WHERE a ~== :a;
             a             
---------------------------
 Euler Taveira de Oliveira
 EULER TAVEIRA DE OLIVEIRA
 Euler T. de Oliveira
 Euler Oliveira
 Euler Taveira
 EULER TAVEIRA OLIVEIRA
 Oliveira, Euler
(7 rows)

-- Cosine
SHOW pg_similarity.cosine_threshold;
 pg_similarity.cosine_threshold 
--------------------------------
 0.7
(1 row)

SELECT a FROM simtst WHERE a ~## :a;
             a             
---------------------------
 Euler Taveira de Oliveira
 Euler T. de Oliveira
 Euler Oliveira
 Euler Taveira
 Oliveira, Euler
(5 rows)

SET pg_similarity.cosine_threshold to 0.9;
SHOW pg_similarity.cosine_threshold;
 pg_similarity.cosine_threshold 
--------------------------------
 0.9
(1 row)

SELECT a FROM simtst WHERE a ~## :a;
             a             
---------------------------
 Euler Taveira de Oliveira
(1 row)

-- Overlap Coefficient
SHOW pg_similarity.overlap_tokenizer;
 pg_similarity.overlap_tokenizer 
---------------------------------
 alnum
(1 row)

SELECT a FROM simtst WHERE a ~** :a;
             a             
---------------------------
 Euler Taveira de Oliveira
 Euler T. de Oliveira
 Euler Oliveira
 Euler Taveira
 Oliveira, Euler
(5 rows)

SET pg_similarity.overlap_tokenizer to 'gram';
SET pg_similarity.overlap_threshold to 0.8;
SELECT a FROM simtst WHERE a ~** :a;
             a             
---------------------------
 Euler Taveira de Oliveira
 Euler T. de Oliveira
 Euler Oliveira
 Euler Taveira
(4 rows)

DROP TABLE simtst;
